utils/sparse.py

```diff
@@ -1,11 +1,44 @@
 import math
 
+from sparsezoo import Zoo
 from sparseml.pytorch.optim import ScheduledModifierManager
 from sparseml.pytorch.utils import SparsificationGroupLogger
 
 from utils.torch_utils import is_parallel
 
 
+def _get_model_framework_file(model, path):
+    transfer_request = 'recipe_type=transfer' in path
+    checkpoint_available = any([file.checkpoint for file in model.framework_files])
+    final_available = any([not file.checkpoint for file in model.framework_files])
+
+    if transfer_request and checkpoint_available:
+        # checkpoints are saved for transfer learning use cases,
+        # return checkpoint if avaiable and requested
+        return [file for file in model.framework_files if file.checkpoint][0]
+    elif final_available:
+        # default to returning final state, if available
+        return [file for file in model.framework_files if not file.checkpoint][0]
+
+    raise ValueError(f"Could not find a valid framework file for {path}")
+
+
+def check_load_sparsezoo_weights(path):
```

Comment:
nit: can we change `load` to "download", so this isn't confused with the action of "loading" torch weights into a module

Reply:
changed